includes/gateway/class-omise-payment-promptpay.php

```diff
@@ -0,0 +1,207 @@
+					}
+
+					let watch = function( duration, display ) {
+						let timer = duration, minutes, seconds;
+							timeInterval = setInterval( function () {
+								minutes = parseInt( timer / 60, 10 )
+								seconds = parseInt( timer % 60, 10 );
+
+								minutes = minutes < 10 ? "0" + minutes : minutes;
+								seconds = seconds < 10 ? "0" + seconds : seconds;
+
+								display.text( minutes + ":" + seconds );
+
+								if ( --timer < 0 ) { clearInterval(timeInterval); }
+							}, 1000);
+					}
+
+					let elementRefreshButton         = $( '#omise-offline-payment-refresh-status' ),
+					    elementOmiseOfflineDetails   = $( '#omise-offline-additional-details' ),
+					    elementTimeoutMessage        = $( '#omise-offline-payment-timeout' ),
+					    elementPaymentResult         = $( '#omise-offline-payment-result' ),
+					    validatePaymentResultCount   = 0;
+					    validatePaymentResultTimerId = null,
+					    intervalTime                 = 10000; // 10 seconds
+					    maxIntervalTime              = 600000; // 10 minutes
+					    validatePaymentResult        = function() {
+							validatePaymentResultCount++;
+
+							$.ajax({
+								type: 'POST',
+								url: '<?php echo admin_url( 'admin-ajax.php' ); ?>',
+								dataType: 'json',
+								data: {
+									'action': 'fetch_order_status',
+									'order_id': '<?php echo $order; ?>'
+								},
+								success: function( response ) {
+									if ( 'processing' === response.data.order_status ) {
+										elementOmiseOfflineDetails.fadeOut();
+										elementPaymentResult.html('<h4><?php echo __( "Payment successful", "omise" ); ?></h4><p><?php echo __( "Thank you. Your order has been paid.", "omise" ); ?></p>').fadeIn();
+										clearInterval( validatePaymentResultTimerId );
+										return;
+									} else if ( 'failed' === response.data.order_status ) {
+										elementOmiseOfflineDetails.fadeOut();
+										elementPaymentResult.html('<h4><?php echo __( "Payment failed", "omise" ); ?></h4><p><?php echo __( "Please try placing an order with a different payment method.", "omise" ); ?></p>').fadeIn();
+										clearInterval( validatePaymentResultTimerId );
+										return;
+									}
+								},
+								error: function( response ) { console.log(response); },
+								complete: function (response ) {
+									if ( validatePaymentResultCount * intervalTime >= maxIntervalTime ) {
+										elementTimeoutMessage.fadeIn();
+										console.log('stop');
+										return;
+									}
+								}
+							});
+						};
+
+					elementRefreshButton.click( function() {
+						validatePaymentResultCount   = 0;
+						validatePaymentResultTimerId = initializeTimer();
+
+						elementTimeoutMessage.fadeOut();
+					} );
+
+					$( document ).ready( function() {
+						validatePaymentResultCount   = 0;
+						validatePaymentResultTimerId = initializeTimer();
+					});
+				})(jQuery);
+			</script>
+		<?php elseif ( 'email' === $context ) : ?>
```

Comment:
This part should be excluded in "Order failed" emails which usually gets send after failed charge creation.

Reply:
This part of the code will be triggered only when the charge is successfully created and user redirects to the thank-you page (or did you get another different result?)

Reply:
Fixed
<img width="800" alt="Screen Shot 2563-08-20 at 01 18 36" src="https://user-images.githubusercontent.com/2154669/90674994-19861700-e284-11ea-8aba-71eaf21ffa8e.png">

Thank a lot! I totally missed this point 👍